### PR TITLE
w5500: Change mac address at runtime

### PR DESCRIPTION
**Pull request by jancoow**
Add ability to change mac address at runtime using net mgmt config call

### CODE REVIEW

File: drivers/ethernet/eth_w5500.c [hunk 1]
```diff
@@ -340,6 +340,17 @@ static int w5500_set_config(const struct device *dev,
 			    enum ethernet_config_type type,
 			    const struct ethernet_config *config)
 {
+	struct w5500_runtime *ctx = dev->data;
+
+	switch (type) {
+	case ETHERNET_CONFIG_TYPE_MAC_ADDRESS:
```
Comment (parthitce) [inline drivers/ethernet/eth_w5500.c:346]: Thanks for changes. Could you please adapt for `ETHERNET_CONFIG_TYPE_PROMISC_MODE` as well with the case statement?

**Review by parthitce — CHANGES_REQUESTED** [review]

**Comment by jancoow** [comment]
@parthitce I've added the PROMISC_MODE in the switch case statement as well

**Comment by GeorgeCGV** [comment]
A similar PR: #48763

**Review by tbursztyka — APPROVED** [review]

**Comment by rlubos** [comment]
@parthitce Can you please revist the PR? It seems that your request has been addressed.

**Comment by parthitce** [comment]
> @parthitce Can you please revist the PR? It seems that your request has been addressed.

Yeah I missed that. LGTM.

**Review by parthitce — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Comment by carlescufi** [comment]
I am going to close this PR to avoid it showing all the time in the "approved" list. @jancoow please reopen this once you have rebased it.

**Comment by carlescufi** [comment]
> I am going to close this PR to avoid it showing all the time in the "approved" list. @jancoow please reopen this once you have rebased it.

You know, I will try to rebase it myself first

**Pull request closed without merging**